File: Java/기본 수학2/직각삼각형.java
import java.io.*;
import java.util.StringTokenizer;
import java.math.*;
import java.util.ArrayList;

public class Main{
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        while(true) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			ArrayList<Integer> num_list = new ArrayList<Integer>(3);
		    for(int i = 0; i < 3; i ++) num_list.add(Integer.parseInt(st.nextToken()));
		    
		    if(num_list.get(0) + num_list.get(1) + num_list.get(2) == 0) break;
		    int max_a = num_list.get(0) >= num_list.get(1) ? num_list.get(0) :
		    	(num_list.get(1) >= num_list.get(2) ? num_list.get(1) : num_list.get(2));
		    num_list.remove(num_list.indexOf(max_a));
		    if(Math.pow(max_a, 2) == Math.pow(num_list.get(0), 2) + Math.pow(num_list.get(1), 2)) System.out.println("right");
		    else System.out.println("wrong");
		}
    }
}

File: C++/기본 수학2/직각삼각형.cc
#include <iostream>
using namespace std;

int main() {
    int a, b, c, t;
    cin >> a >> b >> c;
    
    while(a != 0) {
        if(a > c) {
            t = c;
            c = a;
            a = t;
        }
        if(b > c) {
            t = c;
            c = b;
            b = t;
        }
        
        if(c*c == a*a + b*b)
            cout << "right" << '\n';
        else
            cout << "wrong" << '\n';
        
        cin >> a >> b >> c;
    }
    
    return 0;
}